### PR TITLE
feat: allow sanitizers on GCC as well

### PR DESCRIPTION
**Pull request by Swiftb0y**
`-fsanitize=address/undefined/thread` are all available on GCC too so I see no reason for the restriction to clang.

### CODE REVIEW

**Comment by Holzhaus** [comment]
Thanks.

**Comment by Swiftb0y** [comment]
Thank you.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2773,26 +2773,26 @@ if(CLANG_COLORDIAG)
...
+if(SANITIZE_UNDEFINED)
+  list(APPEND SANITIZERS "undefined")
+endif()
+option(SANITIZE_THREAD "Clang Thread Sanitizer" OFF)
+if(SANITIZE_THREAD)
+  list(APPEND SANITIZERS "thread")
+endif()
+if(NOT SANITIZERS STREQUAL "")
+  if(NOT (LLVM_CLANG OR GNU_GCC))
+    message(FATAL_ERROR "Sanitizers are only available on Clang or GCC")
```
Comment (JoergAtGithub) [inline CMakeLists.txt:2791]: ASAN is also available in MSVC: https://learn.microsoft.com/en-us/cpp/sanitizers/asan

Comment (Swiftb0y) [reply CMakeLists.txt:2791]: yes, but the commandline syntax is a little different and I didn't want to go KISS (also I would have had no way to test the MSVC code path). Feel free to implement yourself in a follow-up PR though.

**Pull request merged**